packages/@webex/plugin-cc/test/unit/spec/services/task/TaskManager.ts

```diff
@@ -0,0 +1,115 @@
+import 'jsdom-global/register';
+import EventEmitter from 'events';
+import { LoginOption } from '../../../../../src/types';
+import { LINE_EVENTS } from '@webex/calling';
+import { CC_EVENTS } from '../../../../../src/services/config/types';
+import TaskManager from '../../../../../src/services/task/TaskManager';
+import * as contact from '../../../../../src/services/task/contact'
+import { TaskData } from '../../../../../src/services/task/types';
+import Task from '../../../../../src/services/task';
+
+jest.mock('./../../../../../src/services/task', () => {
+  return jest.fn().mockImplementation(() => {
+    return {
+      // Optionally mock methods or properties on the Task instance here
+    };
+  });
+});
+
+describe('TaskManager', () => {
+  let taskManager;
+  let contactMock;
+  let webCallingServiceMock ;
+  let webSocketManagerMock;
+
+  beforeEach(() => {
+    contactMock = contact;
+    webCallingServiceMock = new EventEmitter();
+    webSocketManagerMock = new EventEmitter();
+
+    webCallingServiceMock.loginOption = LoginOption.BROWSER;
+
+    taskManager = new TaskManager(contactMock, webCallingServiceMock, webSocketManagerMock);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should initialize TaskManager and register listeners', () => {
+    expect(taskManager).toBeInstanceOf(TaskManager);
+    expect(webCallingServiceMock.listenerCount(LINE_EVENTS.INCOMING_CALL)).toBe(1);
+    expect(webSocketManagerMock.listenerCount('message')).toBe(1);
+  });
+
+  it('should handle incoming call event and emit TASK_INCOMING', () => {
+    const callMock = jest.fn();
+
+    taskManager.on('task:incoming', (task) => {
+      expect(task).toBe(taskManager.task);
```

Comment:
We should be checking this using the getTask getter. The taskManager.task object is a temporary one which will change every time a new event comes in.